### ISSUE TITLE
Feature request: byte offset in conduitParser

### ISSUE THREAD

**Issue by beojan** [snoyberg/conduit#331]
It would be rather nice if `conduitParser` also returned a byte-offset instead of just line and column number. With very large files, the byte-offset would let you `seek` to the appropriate position instead of having to read the file to that point to skip lines.

**Comment by beojan** [comment]
I used [conduit-tokenize-attoparsec](haskell-works/conduit-tokenize-attoparsec) instead, since that provides this functionality. It may be a good implementation to include in conduit-extra, though it doesn't allow tracking *both* byte offset *and* line / column position.

With that version, I found that 
```haskell
event <- do
            conduitParser (Offset 0) (skipMany $ endOfLine <|> headerLine) .| sinkNull
            conduitParser (Offset 0) eventParser
```
would hang forever. Should this do what I expect (discard blank and header lines, then parse an event, returning the position of the event) or am I going wrong somewhere?

**Comment by beojan** [comment]
I see it can't do what I expected because if a chunk contains multiple events it will be repeatedly fed to `eventParser`.

**Comment by snoyberg** [comment]
Even though the change is a breaking change (which I normally avoid like the plague), it's a minor change. I've pushed a new branch (331-offset). The tests aren't passing yet, and I'm about 95% certain I've got off-by-one errors in there. Unfortunately other things have got my attention for the next two weeks, but if you're able to get the tests passing and send a PR, I'd be OK with the change.

**Comment by beojan** [comment]
Done. Turns out it's the *tests* that needed to be fixed.

**Comment by snoyberg** [comment]
Fixed with #332.
